diff --git a/games/encoder/GraphEncoder.py b/games/encoder/GraphEncoder.py
index 436b348..e13204a 100644
--- a/games/encoder/GraphEncoder.py
+++ b/games/encoder/GraphEncoder.py
@@ -210,7 +210,7 @@ class GraphEncoderFreeway:
 
             for i in car_indices:
                 for j in lane_indices:
-                    if abs(node_features[i, 1] - node_features[j, 1]) <= proximity_threshold:
+                    if node_features[i, 1] == node_features[j, 1] :
                         atom_features = torch.zeros((2, object_feature_length)).tolist()
                         graph.add_node(atom_index, type="CarOnLane", features=atom_features)
                         graph.add_edge(i, atom_index, position=0)
diff --git a/games/encoder/__pycache__/GraphEncoder.cpython-310.pyc b/games/encoder/__pycache__/GraphEncoder.cpython-310.pyc
index 3dc7f23..6a2f0b5 100644
Binary files a/games/encoder/__pycache__/GraphEncoder.cpython-310.pyc and b/games/encoder/__pycache__/GraphEncoder.cpython-310.pyc differ
diff --git a/games/freeway/__pycache__/run_supervised_gnn.cpython-310.pyc b/games/freeway/__pycache__/run_supervised_gnn.cpython-310.pyc
index ef42b16..ac3e969 100644
Binary files a/games/freeway/__pycache__/run_supervised_gnn.cpython-310.pyc and b/games/freeway/__pycache__/run_supervised_gnn.cpython-310.pyc differ
diff --git a/games/freeway/freeway_envs/__pycache__/freeway_env.cpython-310.pyc b/games/freeway/freeway_envs/__pycache__/freeway_env.cpython-310.pyc
index 2093cbd..f6d0690 100644
Binary files a/games/freeway/freeway_envs/__pycache__/freeway_env.cpython-310.pyc and b/games/freeway/freeway_envs/__pycache__/freeway_env.cpython-310.pyc differ
diff --git a/games/freeway/freeway_envs/freeway_env.py b/games/freeway/freeway_envs/freeway_env.py
index 7e400ca..31ad478 100644
--- a/games/freeway/freeway_envs/freeway_env.py
+++ b/games/freeway/freeway_envs/freeway_env.py
@@ -72,6 +72,7 @@ class FreewayEnv(gym.Env):
                       'speed': random.randint(1, 2)} for _ in range(self.max_cars)]
         self.done = False
         self.episode_start_time = pygame.time.get_ticks()
+        self.current_steps = 0
         self.frame_buffer = np.zeros((self.frame_stack, 84, 84), dtype=np.uint8)
         if self.observation_type == "pixel":
             for _ in range(self.frame_stack):
@@ -82,13 +83,15 @@ class FreewayEnv(gym.Env):
 
     def step(self, action):
         reward = 0
-        reward = -0.5
+        self.current_steps += 1
+        #reward = -0.5
         current_time = pygame.time.get_ticks()
         if action == 1:  # Up
             self.player_rect.y = max(0, self.player_rect.y - 5)
         elif action == 2:  # Down
             self.player_rect.y = min(self.window_height - self.player_height, self.player_rect.y + 5)
-
+        if self.current_steps % 5000 == 0:
+            self.done = True
         for car in self.cars:
             car['x'] += car['speed']
             if car['x'] > self.window_width:
@@ -102,8 +105,8 @@ class FreewayEnv(gym.Env):
             self.player_rect.y = self.window_height - self.player_height - 10
         
             self.last_time = current_time
-        if current_time - self.episode_start_time >= 60000:  # 60000 milliseconds = 1 minute
-            self.done = True
+        # if current_time - self.episode_start_time >= 60000:  # 60000 milliseconds = 1 minute
+        #     self.done = True
             
         if self.player_rect.y <= 0:  # Reached top
             self.score +=1
diff --git a/games/model/__pycache__/hetero_gnn.cpython-310.pyc b/games/model/__pycache__/hetero_gnn.cpython-310.pyc
index 7853617..1fafc04 100644
Binary files a/games/model/__pycache__/hetero_gnn.cpython-310.pyc and b/games/model/__pycache__/hetero_gnn.cpython-310.pyc differ
diff --git a/games/model/__pycache__/hetero_message_passing.cpython-310.pyc b/games/model/__pycache__/hetero_message_passing.cpython-310.pyc
index 243e3b4..d03f7d3 100644
Binary files a/games/model/__pycache__/hetero_message_passing.cpython-310.pyc and b/games/model/__pycache__/hetero_message_passing.cpython-310.pyc differ
diff --git a/games/model/__pycache__/policy.cpython-310.pyc b/games/model/__pycache__/policy.cpython-310.pyc
index 18a2c62..e375957 100644
Binary files a/games/model/__pycache__/policy.cpython-310.pyc and b/games/model/__pycache__/policy.cpython-310.pyc differ
diff --git a/wandb/debug-internal.log b/wandb/debug-internal.log
index 0355c64..9c08dc0 120000
--- a/wandb/debug-internal.log
+++ b/wandb/debug-internal.log
@@ -1 +1 @@
-run-20240626_173531-u9sj3nyk/logs/debug-internal.log
\ No newline at end of file
+run-20240804_015048-5ov679jb/logs/debug-internal.log
\ No newline at end of file
diff --git a/wandb/debug.log b/wandb/debug.log
index c7c35e9..48c4082 120000
--- a/wandb/debug.log
+++ b/wandb/debug.log
@@ -1 +1 @@
-run-20240626_173531-u9sj3nyk/logs/debug.log
\ No newline at end of file
+run-20240804_015048-5ov679jb/logs/debug.log
\ No newline at end of file
diff --git a/wandb/latest-run b/wandb/latest-run
index ae14d51..611a0d3 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20240626_173531-u9sj3nyk
\ No newline at end of file
+run-20240804_015048-5ov679jb
\ No newline at end of file
